Source/engine/animationinfo.h

```diff
@@ -6,24 +6,43 @@
 #pragma once
 
 #include <stdint.h>
+#include <type_traits>
 
 namespace devilution {
 
 /**
  * @brief Specifies what special logics are applied for a Animation
  */
-enum class AnimationDistributionParams : uint8_t {
-	None,
+enum class AnimationDistributionFlags : uint8_t {
+	None = 0,
 	/*
 	* @brief ProcessAnimation will be called after SetNewAnimation (in same game tick as NewPlrAnim)
 	*/
-	ProcessAnimationPending,
+	ProcessAnimationPending = 1 << 0,
 	/*
 	* @brief Delay of last Frame is ignored (for example, because only Frame and not delay is checked in game_logic)
 	*/
-	SkipsDelayOfLastFrame,
+	SkipsDelayOfLastFrame = 1 << 1,
+	/*
+	* @brief Repeated Animation (for example same player melee attack, that can be repeated directly after hit frame and doesn't need to show all animation frames)
+	*/
+	RepeatedAction = 1 << 2,
 };
 
+inline AnimationDistributionFlags operator&(AnimationDistributionFlags a, AnimationDistributionFlags b)
+{
+	return static_cast<AnimationDistributionFlags>(static_cast<std::underlying_type_t<AnimationDistributionFlags>>(a) & static_cast<std::underlying_type_t<AnimationDistributionFlags>>(b));
+}
+inline AnimationDistributionFlags operator|(AnimationDistributionFlags a, AnimationDistributionFlags b)
+{
+	return static_cast<AnimationDistributionFlags>(static_cast<std::underlying_type_t<AnimationDistributionFlags>>(a) | static_cast<std::underlying_type_t<AnimationDistributionFlags>>(b));
+}
+inline AnimationDistributionFlags& operator|=(AnimationDistributionFlags& a, AnimationDistributionFlags b)
+{
+	a = a | b;
+	return a;
+}
+
```

Comment:
can this be made more general so that it applies to other bitflags as well?

Reply:
Yes for sure! And I would love it. 👍 

That's also why I started the discussion in discord before.
At that time I looked at different implementations, for example:

- https://github.com/grisumbras/enum-flags
  - Doesn't include much overhead
  - Works well with memset (currently needed)
  - Doesn't need a special named last enum value
  - Does this work well with GCC5/6?
  - How would we include this? In 3rdParty folder?
- https://codereview.stackexchange.com/questions/96146/c-flagset-typesafe-usage-of-enumeration-as-bitset-bitmask
  - Doesn't allow bit-flags (even with cast) that aren't allowed
  - Needs to change most Flags from Flag value (1 << 2) to Bit Index (2)
  - Doesn't work for the moment, cause we still use memset (will hopefully change soon)
- and searched for BitfieldOptions but didn't found anything 😢 

My favorite currently would be grisumbras enum-flags. But I thought this topic alone is enough for a own pr, so I tried to reduce the size and focus the topic of this pr.

As I said would love this to be generic.

Possible next steps:

1. Discuss enum flags with this pr
2. Discuss enum flags in another pr/issue
3. If you like grisumbras enum-flags in 3rdParty folder I would create a follow up pr that would introduce this.
4. Something I'm missing 🙂

Reply:
Well if that's your intent then i suggest you use regular old enum for this PR and then we can have a look at the alternatives in a follow up that can then be applied generally.

If the same 13 lines of boiler plate have to be applied manually to each bitflag enum, then I'm not much for it. But if it can be defined in a generic class then it seams perfect.

enum-flags seams to make use of macros which i'm not a fan of.

Reply:
Converted to plain enum. 😉